src/textbooks/textbook-card/TextbooksCard.test.jsx

```diff
@@ -35,4 +69,83 @@ describe('<TextbookCard />', () => {
       expect(getByText(url)).toBeInTheDocument();
     });
   });
+
+  it('render edit TextbookForm after clicking on edit button', async () => {
+    const { getByTestId, queryByTestId } = renderComponent();
+
+    const editButton = getByTestId('textbook-edit-button');
+    userEvent.click(editButton);
+
+    await waitFor(() => {
+      expect(getByTestId('textbook-form')).toBeInTheDocument();
+      expect(queryByTestId('textbook-card')).not.toBeInTheDocument();
+    });
+  });
+
+  it('close edit TextbookForm after clicking on cancel button', async () => {
+    const { getByTestId, queryByTestId } = renderComponent();
+
+    const editButton = getByTestId('textbook-edit-button');
+    userEvent.click(editButton);
+
+    await waitFor(() => {
+      expect(getByTestId('textbook-form')).toBeInTheDocument();
+      expect(queryByTestId('textbook-card')).not.toBeInTheDocument();
+    });
+
+    const cancelButton = getByTestId('cancel-button');
+    userEvent.click(cancelButton);
+
+    await waitFor(() => {
+      expect(queryByTestId('textbook-form')).not.toBeInTheDocument();
+      expect(getByTestId('textbook-card')).toBeInTheDocument();
+    });
+  });
+
+  it('calls onSubmit when the "Save" button is clicked with a valid form', async () => {
+    const { getByPlaceholderText, getByRole, getByTestId } = renderComponent();
+
+    const editButton = getByTestId('textbook-edit-button');
+    userEvent.click(editButton);
+
+    const tabTitleInput = getByPlaceholderText(messages.tabTitlePlaceholder.defaultMessage);
+    const chapterInput = getByPlaceholderText(
+      messages.chapterTitlePlaceholder.defaultMessage.replace('{value}', textbooksMock.textbooks[1].chapters.length),
+    );
+    const urlInput = getByPlaceholderText(messages.chapterUrlPlaceholder.defaultMessage);
+
+    const newFormValues = {
+      tab_title: 'Tab title',
+      chapters: [
+        {
+          title: 'Chapter',
+          url: 'Url',
+        },
+      ],
+      id: textbooksMock.textbooks[1].id,
+    };
+
+    await userEvent.clear(tabTitleInput);
+    userEvent.type(tabTitleInput, newFormValues.tab_title);
+    await userEvent.clear(chapterInput);
+    userEvent.type(chapterInput, newFormValues.chapters[0].title);
+    await userEvent.clear(urlInput);
+    userEvent.type(urlInput, newFormValues.chapters[0].url);
```

Comment:
[nit]: TS80007: await has no effect on the type of this expression.

```suggestion
    userEvent.clear(tabTitleInput);
    userEvent.type(tabTitleInput, newFormValues.tab_title);
    userEvent.clear(chapterInput);
    userEvent.type(chapterInput, newFormValues.chapters[0].title);
    userEvent.clear(urlInput);
    userEvent.type(urlInput, newFormValues.chapters[0].url);
```

Reply:
Fixed